app/assets/javascripts/redux/modules/models/listenSignals.js

```diff
@@ -101,21 +120,37 @@ export const reducer = handleActions({
 
   [LISTEN_SIGNAL_SHOW_REQUEST_SUCCESS]: handlesListenSignalSucccesResponse,
 
-  [LISTEN_SIGNAL_SHOW_REQUEST_FAIL]: (state, action) => ({
-    ...state,
-    data: {
-      ..._.get(state, 'data', {}),
-      [action.meta.id]: {
-        ..._.get(state, `data.${action.meta.id}`, {}),
-        error: action.payload,
-        loading: false,
-        loaded: false,
-      },
-    },
-  }),
+  [LISTEN_SIGNAL_SHOW_REQUEST_FAIL]: handlesListenSignalFailResponse,
+
+  [LISTEN_SIGNALS_POST_REQUEST]: handleSignalRequest,
 
   [LISTEN_SIGNALS_POST_REQUEST_SUCCESS]: handlesListenSignalSucccesResponse,
 
+  [LISTEN_SIGNALS_POST_REQUEST_FAIL]: handlesListenSignalFailResponse,
+
+  [LISTEN_SIGNALS_PUT_REQUEST]: handleSignalRequest,
+
+  [LISTEN_SIGNALS_PUT_REQUEST_SUCCESS]: (state, action) => {
+    const normalizedResponse = normalizeListenSignalResponse(action.payload);
+    const listenSignalResponse = _.get(normalizedResponse, 'entities.listenSignal', {});
+    const id = _.findLastKey(listenSignalResponse);
+    const listenSignal = listenSignalResponse[id];
+
```

Comment:
Yes, you would. You should import the `LISTEN_SIGNAL_PUT_REQUEST` actions (all of them) as well as the `LISTEN_SIGNAL_POST_REQUEST` actions to the responses reducer.